modules/edu.gemini.giapi.client/src/main/scala/edu/gemini/giapi/client/package.scala

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package edu.gemini.giapi.client
+
+import cats._
+import cats.implicits._
+import cats.effect._
+import edu.gemini.aspen.giapi.util.jms.status.StatusGetter
+import edu.gemini.jms.activemq.provider.ActiveMQJmsProvider
+import fs2.async
+
+/**
+  * Represents a connection to a GIAPi based instrument
+  * @tparam F Effect type
+  */
+trait GiapiConnection[F[_]] {
+  def connect: F[Giapi[F]]
+}
+
+/**
+  * Algebra to interact with a GIAPI instrument
+  * @tparam F Effect Type
+  */
+trait Giapi[F[_]] {
+
+  /**
+    * Returns a value for the status item. If not found a `None` is returned
+    */
+  def get[A](statusItem: String): F[Option[A]]
```

Comment:
GIAPI status items are limited to a few types `String`, `Int`, etc. Is there a way I can express that on `A`?